From e8be0aec24791989adae5f311b6a42f82c83dfa7 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:02:10 -0400
Subject: [fs] btrfs: Cleanup the btrfs_workqueue related function type

Message-id: <1407179109-13714-244-git-send-email-sandeen@redhat.com>
Patchwork-id: 87152
O-Subject: [RHEL7.1 PATCH 243/422] btrfs: Cleanup the btrfs_workqueue related function type
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Qu Wenruo <quwenruo@cn.fujitsu.com>

commit 6db8914f9763d3f0a7610b497d44f93a4c17e62e
Author: Qu Wenruo <quwenruo@cn.fujitsu.com>
Date:   Thu Mar 6 04:19:50 2014 +0000

    btrfs: Cleanup the btrfs_workqueue related function type

    The new btrfs_workqueue still use open-coded function defition,
    this patch will change them into btrfs_func_t type which is much the
    same as kernel workqueue.

    Signed-off-by: Qu Wenruo <quwenruo@cn.fujitsu.com>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/async-thread.c b/fs/btrfs/async-thread.c
index a709585..d8c07e5 100644
--- a/fs/btrfs/async-thread.c
+++ b/fs/btrfs/async-thread.c
@@ -255,9 +255,9 @@ static void normal_work_helper(struct work_struct *arg)
 }
 
 void btrfs_init_work(struct btrfs_work *work,
-       void (*func)(struct btrfs_work *),
-       void (*ordered_func)(struct btrfs_work *),
-       void (*ordered_free)(struct btrfs_work *))
+       btrfs_func_t func,
+       btrfs_func_t ordered_func,
+       btrfs_func_t ordered_free)
 {
  work->func = func;
  work->ordered_func = ordered_func;
diff --git a/fs/btrfs/async-thread.h b/fs/btrfs/async-thread.h
index 08d7174..0a891cd 100644
--- a/fs/btrfs/async-thread.h
+++ b/fs/btrfs/async-thread.h
@@ -23,11 +23,13 @@
 struct btrfs_workqueue;
 /* Internal use only */
 struct __btrfs_workqueue;
+struct btrfs_work;
+typedef void (*btrfs_func_t)(struct btrfs_work *arg);
 
 struct btrfs_work {
- void (*func)(struct btrfs_work *arg);
- void (*ordered_func)(struct btrfs_work *arg);
- void (*ordered_free)(struct btrfs_work *arg);
+ btrfs_func_t func;
+ btrfs_func_t ordered_func;
+ btrfs_func_t ordered_free;
 
  /* Don't touch things below */
  struct work_struct normal_work;
@@ -37,13 +39,13 @@ struct btrfs_work {
 };
 
 struct btrfs_workqueue *btrfs_alloc_workqueue(char *name,
-           int flags,
-           int max_active,
-           int thresh);
+           int flags,
+           int max_active,
+           int thresh);
 void btrfs_init_work(struct btrfs_work *work,
-       void (*func)(struct btrfs_work *),
-       void (*ordered_func)(struct btrfs_work *),
-       void (*ordered_free)(struct btrfs_work *));
+       btrfs_func_t func,
+       btrfs_func_t ordered_func,
+       btrfs_func_t ordered_free);
 void btrfs_queue_work(struct btrfs_workqueue *wq,
         struct btrfs_work *work);
 void btrfs_destroy_workqueue(struct btrfs_workqueue *wq);
-- 
1.7.1